### PR TITLE
docs: Update executors docs for 3.3 release

### PR DESCRIPTION
**Pull request by blkperl**
Signed-off-by: William Van Hevelingen <william.vanhevelingen@acquia.com>

Don't bother creating a PR until you've done this:

* [x] Run `make pre-commit -B` to fix codegen, lint, and commit message problems.

Create your PR as a draft.

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it
  does not need to pass.
* Once required tests have passed, you can make it "Ready for review".
* Say how how you tested your changes. If you changed the UI, attach screenshots.

Tips:

* If changes were requested, and you've made them, then dismiss the review to get it looked at again.
* Add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md) if you like.
* You can ask for help!

### CODE REVIEW

**Comment by blkperl** [comment]
@alexec Let me know if you would like other changes to the doc.

File: docs/workflow-executors.md [hunk 1]
```diff
@@ -4,16 +4,55 @@ A workflow executor is a process that conforms to a specific interface that allo
...
+* Configuration:
+  * `command` must be specified for containers.
+
+You can determine the command and args as follows:
+
+```bash
+docker image inspect -f '{{.Config.Entrypoint}} {{.Config.Cmd}}' argoproj/argosay:v2
+```
+
+[Learn more about command and args](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes
```
Comment (alexec) [inline docs/workflow-executors.md:36]: new line?

**Review by alexec — APPROVED** [review]

File: docs/workflow-executors.md [hunk 2]
```diff
@@ -4,16 +4,55 @@ A workflow executor is a process that conforms to a specific interface that allo
...
+## Emissary (emissary)
+
+> v3.1 and after
+
+**default in >= v3.3**
+
+This is the most fully featured executor.
+
+* Reliability:
+  * Works on GKE Autopilot
```
Comment (terrytangyuan) [inline docs/workflow-executors.md:16]: Perhaps worth another discussion but do we want to include vendor-specific information in the docs?

File: docs/workflow-executors.md (hunk 1 above)
Comment (blkperl) [reply docs/workflow-executors.md]: fixed

**Comment by blkperl** [comment]
@alexec are there any plans to deprecate the other executors?

**Comment by alexec** [comment]
No plan to remove them yet.

**Pull request merged**